kotlin/internal/jvm/impl.bzl

```diff
@@ -303,7 +303,7 @@ def kt_jvm_junit_test_impl(ctx):
             direct = ctx.files._java_runtime,
         ),
         # adds common test variables, including TEST_WORKSPACE.
-        testing.TestEnvironment(environment = ctx.attr.env),
+        testing.TestEnvironment(environment = ctx.attr.env, inherited_environment = ctx.attr.env_inherit),
```

Comment:
`inherited_environment` wasn't added until Bazel 5.2.0 and will likely cause issues for anyone using Bazel 5.0.0 to 5.1.1.

Reply:
Actually thinking about this more - this should be safe to merge as people should be using the latest LTS release anyways or have a clear upgrade path by doing a minor `5.x` version bump. What do you think @restingbull @nkoroste?

Reply:
DTM

Reply:
I'm a little bit on: 7 was just pushed, supporting for is reaching.

Folks who need to keep Bazel at for are most likely not updating `rules_kotlin`